tests/acceptance/features/bootstrap/OcmContext.php

```diff
@@ -53,17 +56,43 @@ public function before(BeforeScenarioScope $scope): void {
 	}
 
 	/**
-	 * @When :user generates invitation
-	 * @When :user generates invitation with email :email and description :description
+	 * @return string
+	 */
+	public function getOcisDomain(): string {
+		return $this->extractDomain(\getenv('TEST_SERVER_URL'));
+	}
+
+	/**
+	 * @return string
+	 */
+	public function getFedOcisDomain(): string {
+		return $this->extractDomain(\getenv('TEST_SERVER_FED_URL'));
+	}
+
+	/**
+	 * @param string $url
+	 *
+	 * @return string
+	 */
+	public function extractDomain($url): string {
+		if (!$url) {
+			return "localhost";
+		}
+		return parse_url($url)["host"];
+	}
+
+	/**
+	 * @When :user creates the federation share invitation
+	 * @When :user creates the federation share invitation with email :email and description :description
 	 *
 	 * @param string $user
 	 * @param string $email
 	 * @param string $description
 	 *
-	 * @return void
+	 * @return ResponseInterface
 	 * @throws GuzzleException
 	 */
-	public function userGeneratesInvitation(string $user, $email = null, $description = null): void {
+	public function userCreatesTheFederationShareInvitation(string $user, $email = null, $description = null): ResponseInterface {
```

Comment:
we have to do it similar to `acceptInvitation`.
```suggestion
	public function createInvitation(string $user, $email = null, $description = null): ResponseInterface {
	...
	}
	
	public function userCreatesTheFederationShareInvitation(string $user, $email = null, $description = null): ResponseInterface {
	...
```

Reply:
Yes, I can. but why this implementation is bad?
then we should set `$this->featureContext->setResponse($response);` in the `userCreatesTheFederationShareInvitation` ? and in the `userHasCreatedTheFederationShareInvitation`? correct?

and write down `$this->invitationToken` in the both methods

Reply:
We should avoid re-using stepDef methods in other stepDefs (here: When is used in Given). What we should do is create a helper method (e.g. createInvitation), save any values if need (here `invitationToken`) and return the response but NOT save using `this->featureContext->setResponse($response);`
Then in the stepDefs we should call the helper method, get the response and save the response in case of When step. in Given we should not save the response but assert that the response is good.

Reply:
This is to prevent altering the saved response (in When step) by Given and Then steps. In short, we should only save the response in When steps